sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosConflictsTest.java

```diff
@@ -0,0 +1,457 @@
+import com.azure.cosmos.implementation.RxDocumentClientImpl;
+import com.azure.cosmos.implementation.TestConfigurations;
+import com.azure.cosmos.implementation.directconnectivity.ReflectionUtils;
+import com.azure.cosmos.models.ConflictResolutionPolicy;
+import com.azure.cosmos.models.CosmosConflictProperties;
+import com.azure.cosmos.models.CosmosConflictRequestOptions;
+import com.azure.cosmos.models.CosmosContainerProperties;
+import com.azure.cosmos.models.CosmosItemRequestOptions;
+import com.azure.cosmos.models.CosmosItemResponse;
+import com.azure.cosmos.models.CosmosQueryRequestOptions;
+import com.azure.cosmos.models.CosmosStoredProcedureProperties;
+import com.azure.cosmos.models.FeedResponse;
+import com.azure.cosmos.models.PartitionKey;
+import com.azure.cosmos.models.ThroughputProperties;
+import com.azure.cosmos.rx.TestSuiteBase;
+import org.apache.commons.io.IOUtils;
+import org.assertj.core.util.Strings;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import reactor.core.publisher.Mono;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+
+public class CosmosConflictsTest extends TestSuiteBase {
+    private static final int CONFLICT_TIMEOUT = 120000;
+    private static Logger logger = LoggerFactory.getLogger(CosmosConflictsTest.class);
+    private String sprocBody;
+    private CosmosAsyncClient globalClient;
+    private List<CosmosAsyncClient> regionalClients;
+
+    @BeforeClass(groups = {"multi-master"}, timeOut = SETUP_TIMEOUT)
+    public void before_ConflictTests() throws Exception {
+        sprocBody = IOUtils.toString(
+            getClass().getClassLoader().getResourceAsStream("conflict-resolver-sproc"), "UTF-8");
+        globalClient = new CosmosClientBuilder()
+            .endpoint(TestConfigurations.HOST)
+            .key(TestConfigurations.MASTER_KEY)
+            .contentResponseOnWriteEnabled(true)
+            .endpointDiscoveryEnabled(false)
+            .directMode()
+            .buildAsyncClient();
+
+        GlobalEndpointManager globalEndpointManager =
+            ReflectionUtils.getGlobalEndpointManager((RxDocumentClientImpl) globalClient.getContextClient());
+        DatabaseAccount databaseAccount = globalEndpointManager.getLatestDatabaseAccount();
+        Iterator<DatabaseAccountLocation> locationIterator = databaseAccount.getWritableLocations().iterator();
+        regionalClients = new ArrayList<>();
+        while (locationIterator.hasNext()) {
+            DatabaseAccountLocation accountLocation = locationIterator.next();
+            CosmosAsyncClient regionalClient = new CosmosClientBuilder()
+                .endpoint(accountLocation.getEndpoint())
```

Comment:
you shouldn't pass the regional account endpoint to the CosmosClientBuilder.

instead you should pass the global account endpoint and set the preferred region to be the local region of interest.

Reply:
Done